driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1448,119 +1448,125 @@ private Manager(
     // on it so synchronized is good enough.
     synchronized void init() {
       checkNotClosed(this);
-      if (isInit) return;
+      if (isInit) {
+        if (initException == null) {
+          return;
+        }
+        throw new IllegalStateException(
+            "Error during cluster initialization, please close and retry");
```

Comment:
Should make sure to include the cause (i.e. `throw new IllegalStateException(..., initException)`.

Also the message should be clarified because the user shouldn't need to call `close()`, we should do that for them.  It should instead say something like `Error encountered during cluster initialization, please create a new Cluster instance`.

Reply:
I would even add: "_This cluster has been closed due to an_ error encountered during its initialization, please create a new Cluster instance.".

Reply:
:+1:, That's even better, makes it clear that it was already closed and errored out, not that the current activity caused it to error out.